File: gestao-spring/src/main/java/com/erp/gestao/dto/filterDTO/PessoaFilterDTO.java
package com.erp.gestao.dto.filterDTO;

import com.erp.gestao.enums.AtivoInativoEnum;
import com.erp.gestao.enums.TipoPessoaEnum;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
public class PessoaFilterDTO {

    private Integer id;

    private String nome;

    private String documento;

    private LocalDate nascimento;

    private LocalDate nascimentoInicio;

    private LocalDate nascimentoFim;

    private AtivoInativoEnum situacao;

    private TipoPessoaEnum tipoPessoa;

}

File: gestao-angular/src/app/shared/dto/filterDTO/pessoa.filter.dto.ts
import {AtivoInativoEnum} from "../../enum/ativo.inativo.enum";
import {TipoPessoaEnum} from "../../enum/tipo.pessoa.enum";

export class PessoaFilterDTO {

  id: number;

  nome: string;

  documento: string;

  nascimento: Date;

  nascimentoInicio: Date;

  nascimentoFim: Date;

  situacao: AtivoInativoEnum;

  tipoPessoa: TipoPessoaEnum;
}
